src/mbgl/style/source_impl.cpp

```diff
@@ -89,19 +89,34 @@ void Source::Impl::updateTiles(const UpdateParameters& parameters) {
     const Range<uint8_t> zoomRange = getZoomRange();
 
     // Determine the overzooming/underzooming amounts and required tiles.
-    int32_t overscaledZoom = util::coveringZoomLevel(parameters.transformState.getZoom(), type, tileSize);
-    int32_t tileZoom = overscaledZoom;
+    uint32_t overscaledZoom = util::coveringZoomLevel(parameters.transformState.getZoom(), type, tileSize);
+    uint32_t tileZoom = overscaledZoom;
+    uint32_t lowResolutionZoom = 0;
 
     std::vector<UnwrappedTileID> idealTiles;
+    std::vector<UnwrappedTileID> lowResolutionTiles;
+
     if (overscaledZoom >= zoomRange.min) {
-        int32_t idealZoom = std::min<int32_t>(zoomRange.max, overscaledZoom);
+        uint32_t idealZoom = std::min<int32_t>(zoomRange.max, overscaledZoom);
 
         // Make sure we're not reparsing overzoomed raster tiles.
         if (type == SourceType::Raster) {
             tileZoom = idealZoom;
         }
 
         idealTiles = util::tileCover(parameters.transformState, idealZoom);
+
+        // Request lower zoom level tiles (if configure to do so) in an attempt
+        // to show something on the screen faster at the cost of a little of bandwidth.
+        lowResolutionZoom = parameters.fixedPrefetchZoom ?
+            parameters.fixedPrefetchZoom :
+            idealZoom - parameters.dynamicPrefetchZoom;
+
+        lowResolutionZoom = lowResolutionZoom < zoomRange.min ? zoomRange.min : lowResolutionZoom;
+
+        if (lowResolutionZoom < idealZoom) {
+            lowResolutionTiles = util::tileCover(parameters.transformState, lowResolutionZoom);
```

Comment:
In previous implementations (JS), we've called this `"panZoom"`. Not sure if that's a better name though.